sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/JdbcIOTest.java

```diff
@@ -335,6 +335,76 @@ public void testReadWithSchema() {
     pipeline.run();
   }
 
+  @Test
+  public void testReadWithPartitions() {
+    PCollection<TestRow> rows =
+        pipeline.apply(
+            JdbcIO.<TestRow>readWithPartitions()
+                .withDataSourceConfiguration(DATA_SOURCE_CONFIGURATION)
+                .withRowMapper(new JdbcTestHelper.CreateTestRowOfNameAndId())
+                .withCoder(SerializableCoder.of(TestRow.class))
+                .withTable(String.format("(select * from %s) as subq", READ_TABLE_NAME))
```

Comment:
Let's also add a test for reading from a particular table (without subquery).

Reply:
Done